ddtrace/context.py

```diff
@@ -147,21 +145,13 @@ def close_span(self, span):
             # some children. On the other hand, asynchronous web frameworks still expect
             # to close the root span after all the children.
             tracer = getattr(span, '_tracer', None)
-            if tracer and tracer.debug_logging and span._parent is None and not self._is_finished():
-                opened_spans = len(self._trace) - self._finished_spans
-                log.debug('Root span "%s" closed, but the trace has %d unfinished spans:', span.name, opened_spans)
-                spans = [x for x in self._trace if not x._finished]
-                for wrong_span in spans:
+            unfinished_spans = [x for x in self._trace if not x._finished]
```

Comment:
performance wise how does this compare? especially in the case when a trace has 5k+ spans?

Reply:
Same as before… (see line 153 on the left :)

(If this can be 5k+ long, we should probably remove logging altogether)

Reply:
Well, I think I just mean this strategy in general, since we do it down on L172 now as well.

`get()` gets called when every span closes, so we'll loop through every span in the trace after every span closes until they are all finished.

Reply:
(TBH I'm more worried about simplifying code, limiting locking, and making sure we got things 100% right to then optimize as needed rather than keeping local optimization right now.)

Reply:
The question is, how big is the list usually? Alternatively, I can change this to have two lists of opened/closed spans if the list can grow to millions. If it's a list of 100 items, it might not be worth it.

Reply:
I'd say, typically we see <100 spans max per trace, more of edge cases for some to have 100+ or 1000+.

I just want to make sure we are aware of the implications of code changes here, (well, mostly in `get()`), since it gets called for every span closing, then these finish checking comprehensions will get executed 100 times for a trace that has 100 spans.

Again, if there is no major difference in performance from doing this, then this is definitely much easier code to reason about.

Reply:
Thanks for the feedback. No, I think we'll be fine then. I think it's worth simplifying the code — this is unlikely the final simplification, I'm pretty sure there is still a few things down the road where we can simplify and make things faster in the end 🤞